runtime/compiler/z/codegen/J9S390PrivateLinkage.cpp

```diff
@@ -2251,6 +2251,9 @@ TR::S390PrivateLinkage::buildVirtualDispatch(TR::Node * callNode, TR::RegisterDe
       gcPoint = new (trHeapMemory()) TR::S390RRInstruction(TR::InstOpCode::BASR, callNode, RegRA, RegRA, cg());
       gcPoint->setDependencyConditions(preDeps);
 
+      if (unresolvedSnippet)
+          ((TR::S390VirtualUnresolvedSnippet *)unresolvedSnippet)->setIndirectCallInstruction(gcPoint);
+
```

Comment:
Can we add some braces here and use `static_cast`? Also `unresolvedSnippet` is of pointer type so we should be comparing against `!= NULL`.